vapi/simulator/simulator.go

```diff
@@ -475,6 +479,52 @@ func (s *handler) category(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
+	done := func(delay int) {
+		go time.AfterFunc(time.Duration(delay)*time.Minute, func() {
+			s.Lock()
+			s.ShutdownStatus.ShutDownTime = ""
+			s.ShutdownStatus.Action = ""
+			s.ShutdownStatus.Reason = ""
+			s.Unlock()
+		})
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		OK(w, s.ShutdownStatus)
+	case http.MethodPost:
+		switch r.URL.Query().Get("action") {
+		case shutdown.Cancel:
+			var spec shutdown.Spec
+			if s.decode(r, w, &spec) {
+				s.ShutdownStatus.ShutDownTime = ""
+				s.ShutdownStatus.Reason = ""
+				s.ShutdownStatus.Action = ""
+			}
+			OK(w, s.ShutdownStatus)
```

Comment:
IMHO these should move inside the `if` since we only report success if decode suceeds (decode writes BadRequest if decode fails causing HTTP write errors)

Reply:
Also, shutdown VAPI returns `204` not `200`.

Reply:
Added code to handle 204 and 200 separately